File: tests/journal.lib/channel_inventory.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// type aliases
template <typename severityT>
using channel_t = pyre::journal::channel_t<severityT>;


// severity stub
class severity_t : public channel_t<severity_t>
{
    // metamethods
public:
    // index initialization is required...
    inline severity_t(const name_type & name): channel_t<severity_t>(name) {}
};


// verify that the default channel state is what we expect
int main() {
    // make a channel
    severity_t channel("test.channel");

    // verify it is on, by default
    assert (channel.active() == true);
    // not fatal
    assert (channel.fatal() == false);
    // and that its device is whatever is set globally
    assert (channel.device() == pyre::journal::chronicler_t::device());

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/channel_inventory.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


# verify that the default channel state is as expected
def test():
    """
    Exercise the default channel state
    """
    # get the package
    from journal.Channel import Channel

    # derive a severity
    class Severity(Channel, active=False):
        """
        A sample derivation
        """

    # make one
    channel = Severity(name="test.channel")

    # get its inventory
    inventory = channel.inventory
    # verify that it is disabled, as we specified
    assert inventory.active == False
    # and that it is non-fatal, by default
    assert inventory.fatal == False
    # and that it has no registered device
    assert inventory.device == None

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
